python_modules/dagster/dagster/_core/definitions/run_request.py

```diff
@@ -90,19 +100,88 @@ def __new__(
                 asset_selection, "asset_selection", of_type=AssetKey
             ),
             stale_assets_only=check.bool_param(stale_assets_only, "stale_assets_only"),
+            partition_key=check.opt_str_param(partition_key, "partition_key"),
         )
 
-    @property
-    def partition_key(self) -> Optional[str]:
-        return self.tags.get(PARTITION_NAME_TAG)
-
     def with_replaced_attrs(self, **kwargs: Any) -> "RunRequest":
         fields = self._asdict()
         for k in fields.keys():
             if k in kwargs:
                 fields[k] = kwargs[k]
         return RunRequest(**fields)
 
+    def with_resolved_partition(
```

Comment:
Would `with_tags_and_config_from_partition_key` be accurate? If so, I think it would make it more clear what's happening here.

Reply:
yep I think this is a better name